core/domain/collection_domain.py

```diff
@@ -681,23 +804,27 @@ def _convert_collection_contents_v5_dict_to_v6_dict(
         Returns:
             dict. The updated collection_contents dict.
         """
+        # MyPy doesn't allow key deletion from TypedDict, thus we add an ignore.
```

Comment:
Also mention that we can ignore it because we are in the convert function.

Reply:
added a new comment.